internal/storage/storage_handle_test.go

```diff
@@ -109,3 +109,10 @@ func (t *StorageHandleTest) TestNewStorageHandleWhenUserAgentIsSet() {
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
+
+func (t *StorageHandleTest) TestNewStorageHandleWhenJsonReadEnabled() {
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.ExperimentJsonRead = true
+
+	t.invokeAndVerifyStorageHandle(sc)
```

Comment:
can't we trigger an actual read in bucket)_handle unit test and see if json reads are working?

Reply:
We can't test the json read workflow as part of unit test. As client creation part is hardcoded right now and I don't find any option to pass the option WithJsonReads in the fakestorage package.

See here - https://github.com/fsouza/fake-gcs-server/blob/09b807c1d1c02e277d61c48a6ec99e730b7cdf47/fakestorage/server.go#L479

We can test it via integration test, but hard to check, if it's happening with xml/json programmatically. I have added the argument in our codebase to check basic operation.